main.tf

```diff
@@ -64,6 +64,17 @@ module "project-factory" {
   vpc_service_control_perimeter_name = var.vpc_service_control_perimeter_name
 }
 
+/******************************************
+  Setting API service accounts for shared VPC
+ *****************************************/
+module "shared_vpc_access" {
+  source             = "./modules/shared_vpc_access"
+  host_project_id    = var.shared_vpc
```

Comment:
Would we not need additional checks in `shared_vpc_access` to catch the  case if `var.shared_vpc == ""`, which is the default value? I am also concerned if those checks (if implemented with `coun`t) would bring a scenario where TF may complain about being unable to determine number of resources until run time.